modules/javafx.graphics/src/test/java/test/javafx/scene/layout/RegionTest.java

```diff
@@ -1250,34 +1250,102 @@ public void clearDirty() {
+        assertEquals(Long.MAX_VALUE, region.snapSpaceX(Double.MAX_VALUE), 0.0);
+        assertEquals(Long.MAX_VALUE, region.snapSpaceY(Double.MAX_VALUE), 0.0);
+        assertEquals(Long.MIN_VALUE, region.snapSpaceX(-Double.MAX_VALUE), 0.0);
+        assertEquals(Long.MIN_VALUE, region.snapSpaceY(-Double.MAX_VALUE), 0.0);
+
+        assertEquals(Long.MAX_VALUE, region.snapSpaceX(Double.POSITIVE_INFINITY), 0.0);
+        assertEquals(Long.MAX_VALUE, region.snapSpaceY(Double.POSITIVE_INFINITY), 0.0);
+        assertEquals(Long.MIN_VALUE, region.snapSpaceX(Double.NEGATIVE_INFINITY), 0.0);
+        assertEquals(Long.MIN_VALUE, region.snapSpaceY(Double.NEGATIVE_INFINITY), 0.0);
+
+        stage.setRenderScaleX(1.5);
+        stage.setRenderScaleY(1.5);
+
+        // Size functions:
+
+        assertEquals(Double.MAX_VALUE, region.snapSizeX(Double.MAX_VALUE), Math.ulp(Double.MAX_VALUE));
+        assertEquals(Double.MAX_VALUE, region.snapSizeY(Double.MAX_VALUE), Math.ulp(Double.MAX_VALUE));
+        assertEquals(-Double.MAX_VALUE, region.snapSizeX(-Double.MAX_VALUE), 0.0);
+        assertEquals(-Double.MAX_VALUE, region.snapSizeY(-Double.MAX_VALUE), 0.0);
+
+        assertEquals(Double.POSITIVE_INFINITY, region.snapSizeX(Double.POSITIVE_INFINITY), 0.0);
+        assertEquals(Double.POSITIVE_INFINITY, region.snapSizeY(Double.POSITIVE_INFINITY), 0.0);
+        assertEquals(Double.NEGATIVE_INFINITY, region.snapSizeX(Double.NEGATIVE_INFINITY), 0.0);
+        assertEquals(Double.NEGATIVE_INFINITY, region.snapSizeY(Double.NEGATIVE_INFINITY), 0.0);
+
+        // Space functions:
+
+        // These are even more bugged, they divide the long max/min value by scale after using the round instead of rint
+        assertEquals(Long.MAX_VALUE / 1.5, region.snapSpaceX(Double.MAX_VALUE), 0.0);
+        assertEquals(Long.MAX_VALUE / 1.5, region.snapSpaceY(Double.MAX_VALUE), 0.0);
+        assertEquals(Long.MIN_VALUE / 1.5, region.snapSpaceX(-Double.MAX_VALUE), 0.0);
+        assertEquals(Long.MIN_VALUE / 1.5, region.snapSpaceY(-Double.MAX_VALUE), 0.0);
+
+        assertEquals(Long.MAX_VALUE / 1.5, region.snapSpaceX(Double.POSITIVE_INFINITY), 0.0);
+        assertEquals(Long.MAX_VALUE / 1.5, region.snapSpaceY(Double.POSITIVE_INFINITY), 0.0);
+        assertEquals(Long.MIN_VALUE / 1.5, region.snapSpaceX(Double.NEGATIVE_INFINITY), 0.0);
+        assertEquals(Long.MIN_VALUE / 1.5, region.snapSpaceY(Double.NEGATIVE_INFINITY), 0.0);
+    }
+
     // Test for JDK-8255415
-    @Test public void snappingASnappedValueGivesTheSameValueTest() {
+    @Test
+    public void snappingASnappedValueGivesTheSameValueTest() {
         Stage stage = new Stage();
         Region region = new Region();
         Scene scene = new Scene(region);
         stage.setScene(scene);
 
-        double[] scales = new double[] {1.0, 1.25, 1.5, 1.75, 2.0, 1.374562997};
+        double[] scales = new double[] {1.0, 1.25, 1.5, 1.75, 2.0, 1.374562997, 20.0};
+        Random random = new Random();
```

Comment:
In my opinion, using a fixed seed would have been preferable here.

Reply:
I agree with you, but I'll like to leave it now as-is as it was already discussed, and it won't make much of a difference at this point as there are better tests in place directly on `ScaledMath`.